### PR TITLE
fix: invalidating ssr module

### PR DESCRIPTION
**Pull request by Aslemammad**
This avoids hydration mismatch error in DEV.

### CODE REVIEW

File: packages/waku/src/lib/plugins/vite-plugin-rsc-hmr.ts [hunk 1]
```diff
@@ -41,7 +42,7 @@ if (import.meta.hot && !globalThis.__WAKU_HMR_CONFIGURED__) {
     document.head.appendChild(script);
     // avoid HMR flash by first applying the new and removing the old styles 
     if (style) {
-      queueMicrotask(style.remove);
+      queueMicrotask(() => style.parentElement?.removeChild(style));
```
Comment (dai-shi) [inline packages/waku/src/lib/plugins/vite-plugin-rsc-hmr.ts:45]: I think you want this?
```suggestion
      queueMicrotask(() => style.remove());
```

File: packages/waku/src/lib/plugins/vite-plugin-rsc-hmr.ts [hunk 2]
```diff
@@ -122,6 +123,21 @@ export function rscHmrPlugin(): Plugin {
         );
       }
     },
+    handleHotUpdate({ file }) {
+      const moduleLoading = (globalThis as any).__webpack_module_loading__;
+      const moduleCache = (globalThis as any).__webpack_module_cache__;
+      const id = filePathToFileURL(file);
+      if (moduleLoading.has(id) && moduleCache.has(id)) {
```
Comment (dai-shi) [inline packages/waku/src/lib/plugins/vite-plugin-rsc-hmr.ts:130]: Should we check only the first one?
```suggestion
      if (moduleLoading.has(id)) {
```

File: packages/waku/src/lib/plugins/vite-plugin-rsc-hmr.ts (hunk 1 above)
Comment (Aslemammad) [reply packages/waku/src/lib/plugins/vite-plugin-rsc-hmr.ts:45]: Nope this errors out sometimes. Illegal invocation.

File: packages/waku/src/lib/plugins/vite-plugin-rsc-hmr.ts (hunk 2 above)
Comment (Aslemammad) [reply packages/waku/src/lib/plugins/vite-plugin-rsc-hmr.ts:130]: I do not know about react's impl details, but I think checking the two might be safer?

File: packages/waku/src/lib/plugins/vite-plugin-rsc-hmr.ts (hunk 1 above)
Comment (dai-shi) [reply packages/waku/src/lib/plugins/vite-plugin-rsc-hmr.ts:45]: I see...

File: packages/waku/src/lib/plugins/vite-plugin-rsc-hmr.ts (hunk 2 above)
Comment (dai-shi) [reply packages/waku/src/lib/plugins/vite-plugin-rsc-hmr.ts:130]: It's not React impl, but yeah it can be a little safer. Let's leave it.

**Review by dai-shi — APPROVED** [review]
Thanks. This is a nice improvement.

**Pull request merged**